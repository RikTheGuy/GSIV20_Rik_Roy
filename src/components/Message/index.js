import React from 'react'
import './index.scss'

const index = ({ message = '', danger = false }) => {
    return (
        <div data-testid='message' className={`message ${danger && 'danger'}`}>
            {message}
        </div>
    )
}

export default index
