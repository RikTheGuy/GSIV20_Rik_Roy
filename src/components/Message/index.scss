@import "../../styles/colors.scss";

.message {
  display: block;
  padding: 1rem;
  margin: 1rem;
  border: 2px solid $accent;
  color: $accent;
  background-color: $primary;
  border-radius: 0.2rem;
  text-align: center;
  font-weight: 600;

  &.danger {
    color: $red;
  }
}
