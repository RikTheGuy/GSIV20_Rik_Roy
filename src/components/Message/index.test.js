import Message from './index.js';
import { render, cleanup } from '@testing-library/react';

import renderer from 'react-test-renderer';

describe('Message', () => {
    afterEach(cleanup)

    describe('Renders Successfully', () => {
        test('Renders With Message', () => {
            const { getByTestId } = render(<Message message='Sample Message' />)
            expect(getByTestId('message')).toHaveTextContent('Sample Message')
        })

        test('Renders Without Message', () => {
            const { getByTestId } = render(<Message />)
            expect(getByTestId('message')).toHaveTextContent('')
        })

        test('Renders With Error', () => {
            const { getByTestId } = render(<Message message='Sample Message' danger />)
            expect(getByTestId('message')).toHaveTextContent('Sample Message')
            expect(getByTestId('message')).toHaveTextContent('Sample Message')
        })

    })

    describe('Snapshot Testing', () => {
        test('Normal', () => {
            const tree = renderer.create(<Message message='Sample Message' />).toJSON()
            expect(tree).toMatchSnapshot()
        })

        test('Danger', () => {
            const tree = renderer.create(<Message message='Sample Message' danger />).toJSON()
            expect(tree).toMatchSnapshot()
        })
    })
})