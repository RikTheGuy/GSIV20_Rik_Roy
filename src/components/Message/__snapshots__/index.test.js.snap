// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Snapshot Testing Danger 1`] = `
<div
  className="message danger"
  data-testid="message"
>
  Sample Message
</div>
`;

exports[`Message Snapshot Testing Normal 1`] = `
<div
  className="message false"
  data-testid="message"
>
  Sample Message
</div>
`;
